**Task1: Table Booking and Online Delivery**

Task List:

1. Determine the percentage of restaurants that offer table booking and online delivery.

2. Compare the average ratings of restaurants with table booking and those without.

3. Analyze the availability of online delivery among restaurants with different price ranges.

In [2]:
import pandas as pd

In [3]:
new_data = pd.read_csv("new_data.csv")

In [4]:
new_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9551 entries, 0 to 9550
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Restaurant ID         9551 non-null   int64  
 1   Restaurant Name       9551 non-null   object 
 2   Country Code          9551 non-null   int64  
 3   City                  9551 non-null   object 
 4   Address               9551 non-null   object 
 5   Locality              9551 non-null   object 
 6   Locality Verbose      9551 non-null   object 
 7   Longitude             9551 non-null   float64
 8   Latitude              9551 non-null   float64
 9   Cuisines              9542 non-null   object 
 10  Average Cost for two  9551 non-null   int64  
 11  Currency              9551 non-null   object 
 12  Has Table booking     9551 non-null   object 
 13  Has Online delivery   9551 non-null   object 
 14  Is delivering now     9551 non-null   object 
 15  Switch to order menu 

1. Getting the percentage of restaurants that offer table booking and online delivery

In [5]:
# Total number of restaurants
total_number_of_restaurants = new_data.groupby('Restaurant Name').size().reset_index(name='Count')
print(f"Total number of observed unique restaurants: {total_number_of_restaurants.shape[0]}")

# Total number of restaurants offering both table booking and online delivery
restaurants_with_TB_OD = new_data[(new_data['Has Table booking']=='Yes') & (new_data['Has Online delivery']=='Yes')]
restaurants_with_TB_OD_grouped_by_restaurants = restaurants_with_TB_OD.groupby('Restaurant Name').size().reset_index(name='Count')
print(f"Total number of unique restaurants offering both table booking and online delivery: {restaurants_with_TB_OD_grouped_by_restaurants.shape[0]}")

# Percentage of restuarants offering both table booking and online delivery
percentage_of_restaurants_offering_both_TB_and_OD = ((restaurants_with_TB_OD_grouped_by_restaurants.shape[0])/(total_number_of_restaurants.shape[0])) * 100

print(f"Percentage of restaurants offering table booking and online delivery: {round(percentage_of_restaurants_offering_both_TB_and_OD, 2)}%")
# restaurants_with_TB_OD_grouped_by_restaurants['Count'].max()
# restaurants_with_TB_OD_count = restaurants_with_TB_OD.shape[0]
# print(f"There are {restaurants_with_TB_OD_count} number of restaurants that offer table booking and online delivery")

Total number of observed unique restaurants: 7446
Total number of unique restaurants offering both table booking and online delivery: 364
Percentage of restaurants offering table booking and online delivery: 4.89%


2. Comparing the average ratings of restaurants with table booking and those without.


In [6]:
# Getting the average ratings of restaurants with table booking

# first we get the data of all restaurants with table booking 
restaurants_with_TB = new_data[new_data['Has Table booking']=='Yes']

# Getting the average ratings of restaurants with table booking
avg_rating_restaurants_with_TB = restaurants_with_TB.groupby('Restaurant Name')['Aggregate rating'].mean()

# Getting the overall average of the restaurants with table booking
meanOfThe_avg_rating_restaurants_with_TB = avg_rating_restaurants_with_TB.reset_index(name='Aggregate rating')['Aggregate rating'].mean()

meanOfThe_avg_rating_restaurants_with_TB

3.4436431922507875

In [7]:
# Getting the average ratings of restaurants without table booking

# first we get the data of all restuarants with table booking
restaurant_without_TB = new_data[new_data['Has Table booking']=='No']

# Getting the average ratings of restaurants with table booking
avg_rating_restaurants_without_TB = restaurant_without_TB.groupby('Restaurant Name')['Aggregate rating'].mean()

# Getting the overall average of the restaurants without table booking
meanOfThe_avg_rating_restaurants_without_TB = avg_rating_restaurants_without_TB.reset_index(name='Aggregate rating')['Aggregate rating'].mean()
meanOfThe_avg_rating_restaurants_without_TB


2.5093828043448814

Restaurants with table booking have an overall average rating of 3.44 while those without table booking had an overall average rating of 2.51.

This could suggest that the presence of adopting table booking could have had an influence in the rating behaviours of customers as on the average, restaurants having table booking are likely to have high rating compared to those that do not.

**3. Analyzing the availability of online delivery among restaurants with different price ranges**

In [11]:
od_to_pr_check = new_data.groupby('Has Online delivery')['Price range'].value_counts()
od_to_pr_check

Has Online delivery  Price range
No                   1              3743
                     2              1827
                     3               997
                     4               533
Yes                  2              1286
                     1               701
                     3               411
                     4                53
Name: count, dtype: int64

In [12]:
pr_to_od_check = new_data.groupby('Price range')['Has Online delivery'].value_counts()
pr_to_od_check

Price range  Has Online delivery
1            No                     3743
             Yes                     701
2            No                     1827
             Yes                    1286
3            No                      997
             Yes                     411
4            No                      533
             Yes                      53
Name: count, dtype: int64

**The above result suggests the following:**

i.  There are more activities from restaurants with no online delivery irrespective of price range.

ii. Price range 1 has the highest count for restaurants with no online delivery and the second highest count for restaurants with online delivery

iii. Price range 2 has the second highest count for restaurants with no online delivery and the highest count for restaurants with online delivery

iv. Price range 3 has the third highest count for both categories of restaurants with no online delivery and with online delivery

v. Price range 4 has the least count for both categories of reataurants with no online delivery and with online delivery

**Potential Insights:**

i. Price Range Distribution: Restaurants that offer online delivery tend to have lower counts in the more expensive price ranges (3 and 4) compared to those that do not offer online delivery with an exception of price range of 2 of which could be studied further on why it has much activities compared to others.

ii. Market Segmentation: There might be a market trend where more affordable or mid-range price restaurants are more likely to offer online delivery. Conversely, higher-priced restaurants may not have online delivery options as frequently.

iii. Business Strategy: This distribution could reflect different business models or customer preferences. Restaurants with higher price ranges might cater to a different clientele or focus more on in-person dining experiences rather than online delivery.